pkg/pod/pod.go

```diff
@@ -322,6 +327,39 @@ func (b *Builder) Build(ctx context.Context, taskRun *v1beta1.TaskRun, taskSpec
 		return nil, err
 	}
 
+	readonly := true
+	if config.IsSpireEnabled(ctx) {
+		// add SPIRE's CSI volume to the explicitly declared use volumes
+		volumes = append(volumes, corev1.Volume{
+			Name: spire.WorkloadAPI,
+			VolumeSource: corev1.VolumeSource{
+				CSI: &corev1.CSIVolumeSource{
+					Driver:   "csi.spiffe.io",
+					ReadOnly: &readonly,
```

Comment:
why is this handled differently from `ReadOnly` elsewhere in this function?

Reply:
It's because this is a pointer type and Go doesn't let you dereference without a variable reference like `&true`. Another common way i've seen to tackle this is to use something like https://pkg.go.dev/knative.dev/pkg/ptr#Bool and do `ptr.Bool(true)`

Reply:
ah I see, thanks for the explanation @wlynch!

 @jagathprakash maybe the other sections can reuse the variable then to be consistent

Reply:
Done